File: 이진영/DP/등굣길.js
function solution(m, n, puddles) {
    // [m][n] = [m - 1][n] + [m][n - 1]
    let route = Array.from(Array(m), () => Array(n).fill(-1))
    
    puddles.forEach(val => {
        route[val[0] - 1][val[1] - 1] = 0;
    })
    route[0][0] = 1;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (route[i][j] === -1) {
                let sum = 0;
                if (i - 1 >= 0) sum += route[i - 1][j];
                if (j - 1 >= 0) sum += route[i][j - 1];
                route[i][j] = sum % 1000000007;
            }
        }
    }
    
    return route[m - 1][n - 1];
    
}

File: 정민선/DP/등굣길.py
## dp 어렵당..
def solution(m, n, puddles):
    puddles = [[q-1,p-1] for [p,q] in puddles] 
    dp = [[0]*m for _ in range(n)]
    dp[0][0] = 1        

    for i in range(n):
        for j in range(m):
            if [i, j] in puddles: 
                dp[i][j] = 0
            else:       
                if i > 0:
                    dp[i][j] += dp[i - 1][j]
                if j > 0:
                    dp[i][j] += dp[i][j - 1]
                dp[i][j] %= 1000000007

    return dp[n-1][m-1]
